examples/pissa_finetuning/README.md

```diff
@@ -0,0 +1,73 @@
+If the existing settings do not meet your needs, apply [PiSSA initialization](https://github.com/fxmeng/peft/blob/606a69279480bbdea847f4e5247804bdf7e6b898/examples/pissa_finetuning/pissa_finetuning.py#L85-L103) to a pre-trained model and save the decomposed parameters:
+
+```
+# Load an original pre-processed dodel:
+model = AutoModelForCausalLM.from_pretrained(...)
+
+# Configure the initialization method to "pissa", which may take several minutes to execute SVD on the pre-trained model:
+lora_config = LoraConfig(init_lora_weights="pissa", ...) 
+
+# Alternatively, execute fast SVD, which takes only a few seconds. The number of iterations determines the trade-off between the error and computation time:
+# lora_config = LoraConfig(init_lora_weights="pissa_niter_[number of iters]", ...) 
+
+# Perform PiSSA on the original model according to lora_config:
+peft_model = get_peft_model(model, lora_config)
+```
+To eliminate the errors introduced by Fast SVD, we have modified the computation formula for the residual matrix to $W^{res} = W - AB$. Although the calculation of $A$ and $B$ involves errors, the overall initialization error is zero.
+
+### Step 2 (Recommend)
+[Saving the residual model and PiSSA Adapter](https://github.com/fxmeng/peft/blob/51161a52cac3a736d931d90e676b24a32c4f8cd6/src/peft/utils/pissa_utils.py#L27-L51):
+```
+pissa_pre_training_saving(peft_model, saving_path, ...)
+```
+Store the decomposed model locally to prevent the necessity of re-decomposition across multiple fine-tuning experiments, and reducing errors when quantization of the residual model pre-decomposed at full precision.
+
+### Step 3 (Optional)
+If quantization fine-tuning is desired, reload the pre-processed [residual model](https://github.com/fxmeng/peft/blob/606a69279480bbdea847f4e5247804bdf7e6b898/examples/pissa_finetuning/pissa_finetuning.py#L107-L116) in 4-bit or 8-bit configurations along with the full-precision [PiSSA Adapter](https://github.com/fxmeng/peft/blob/606a69279480bbdea847f4e5247804bdf7e6b898/examples/pissa_finetuning/pissa_finetuning.py#L122):
+```
+res_model = AutoModelForCausalLM.from_pretrained(saving_path, load_in_4/8bit=True, ...)
+peft_model = PeftModel.from_pretrained(res_model, f"{saving_path}/pissa_init", is_trainable=True)
+```
+When SVD is conducted at full precision, the PiSSA adapter retains the high-frequency principal components of the original model. 
+Then quantizing the residual model, rather than the original model, notably decreases the quantization error.
+
+### Step 4. 
+[Training](https://github.com/fxmeng/peft/blob/51161a52cac3a736d931d90e676b24a32c4f8cd6/examples/pissa_finetuning/pissa_finetuning.py#L131-L139) the principal singular values and singular vectors results in faster convergence and enhanced performance:
+```
+dataset = load_dataset(...)
+trainer = SFTTrainer(peft_model, dataset, ...)
+peft_model.save_pretrained(os.path.join(args.output_path, "pissa_init"))
```

Comment:
Do we need this, is this different from what we got above when calling `pissa_pre_training_saving`? Or is this just in case that we skipped step 2?